lib/providers/leave_request_provider.dart

```diff
@@ -30,4 +33,37 @@ class LeaveRequestListProvider extends StateNotifier<List<LeaveRequest>> {
   void addLeaveRequest(LeaveRequest leaveRequest) {
     state = [...state, leaveRequest];
   }
+
+  void approveLeaveRequest(LeaveRequest leaveRequest) {
+    state = [
+      for (final request in state)
+        if (request.id == leaveRequest.id)
+          LeaveRequest(
+            id: request.id,
+            user: request.user,
+            viewPermission: request.viewPermission,
+            startDate: request.startDate,
+            endDate: request.endDate,
+            reason: request.reason,
+            leaveType: request.leaveType,
+            status: RequestStatus.approved,
+          )
```

Comment:
For this you can make a `copyWith` methond on `LeaveRequest`: 

`LeaveRequest copyWith({
    String? id,
    String? user,
    ViewPermission? viewPermission,
    DateTime? startDate,
    DateTime? endDate,
    String? reason,
    LeaveType? leaveType,
    RequestStatus? status,
  }) =>
      LeaveRequest(
        id: id ?? this.id,
        endDate: endDate ?? this.endDate,
        leaveType: leaveType ?? this.leaveType,
        reason: reason ?? this.reason,
        startDate: startDate ?? this.startDate,
        status: status ?? this.status,
        user: user ?? this.user,
        viewPermission: viewPermission ?? this.viewPermission,
      );`

Reply:
Then you can write `request.id == leaveRequest.id ? state.copyWith(status: RequestStatus.approved) : request`

Reply:
Also, you can separate the logic of modifying a request in the list into a separate function